catalog/resource_metastore.go

```diff
@@ -87,34 +87,11 @@ func ResourceMetastore() *schema.Resource {
 			}
 			return m
 		})
-	update := func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
-		// other fields to come later
-		updatable := []string{"owner", "name", "delta_sharing_enabled",
-			"delta_sharing_recipient_token_lifetime_in_seconds", "delta_sharing_organization_name"}
-		patch := map[string]interface{}{}
-		for _, field := range updatable {
-			old, new := d.GetChange(field)
-			if old == new {
-				continue
-			}
-			if field == "name" && old == "" {
-				continue
-			}
-			// delta sharing enabled and new is true must always be accompanied by a value for
-			// delta_sharing_recipient_token_lifetime_in_seconds
-			if field == "delta_sharing_enabled" && old != new && new == true &&
-				!d.HasChange("delta_sharing_recipient_token_lifetime_in_seconds") {
-				patch["delta_sharing_recipient_token_lifetime_in_seconds"] =
-					d.Get("delta_sharing_recipient_token_lifetime_in_seconds")
-			}
+	updateOwner := updateFunctionFactory("/unity-catalog/metastores/", []string{"owner", "delta_sharing_enabled",
```

Comment:
```suggestion
	updateOwner := updateFunctionFactory("/unity-catalog/metastores", []string{"owner", "delta_sharing_enabled",
```

nit: can we have it wihtout a trailing slash please?... just looks ugly :(